MeesGame/Gameplay/GameObjects/TileField.cs

```diff
@@ -53,6 +53,31 @@ public TileType GetType(int x, int y)
 
         public void Add(Tile obj, int x, int y)
         {
+            // If You are on the edge, but not on a corner, you can place and EndTile
+            if (OnEdgeOfTileField(x, y) && !OnCornerOfTileField(x, y) && obj.TileType == TileType.End)
+            {
+                base.Add(obj, x, y);
```

Comment:
Please rework this section. Do not change the tile if the selected tile isn't allowed. Don't use more than one base.Add(). Simply return if the tile isn't allowed.